File: Android/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/info/WeexInfoHacker.java
package com.didichuxing.doraemonkit.weex.info;


import org.apache.weex.WXEnvironment;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Set;

/**
 * @author haojianglong
 * @date 2019-06-25
 */
public class WeexInfoHacker {

    public static List<WeexInfo> getWeexInfos() {
        List<WeexInfo> infos = new ArrayList<>();
        Set<Map.Entry<String, String>> entrySet = WXEnvironment.getConfig().entrySet();
        for (Map.Entry<String, String> entry : entrySet) {
            infos.add(new WeexInfo(entry.getKey(), entry.getValue()));
        }
        return infos;
    }

}

File: Android/kotlin/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/info/WeexInfoHacker.kt
package com.didichuxing.doraemonkit.weex.info

import org.apache.weex.WXEnvironment

/**
 * Transformed by alvince on 2020/6/30
 *
 * @author haojianglong
 * @date 2019-06-25
 */
object WeexInfoHacker {

    fun getWeexInfos(): List<WeexInfo> = mutableListOf<WeexInfo>()
        .apply {
            WXEnvironment.getConfig()
                .filter { entry -> !entry.key.isNullOrEmpty() }
                .map { entry -> WeexInfo(entry.key, entry.value ?: "") }
                .forEach { info -> add(info) }
        }

}
